packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue

```diff
@@ -920,10 +922,19 @@ export default defineComponent({
 		): Promise<ICredentialsResponse | null> {
 			let credential;
 			try {
-				credential = await this.credentialsStore.updateCredential({
-					id: this.credentialId,
-					data: credentialDetails,
-				});
+				if (this.credentialPermissions.update) {
+					credential = await this.credentialsStore.updateCredential({
+						id: this.credentialId,
+						data: credentialDetails,
+					});
+				}
+				if (this.isSharedWithChanged && credentialDetails.sharedWith) {
```

Comment:
I meant to ask if it's worth adding a check for the `credential:share` scope, not removing. Sorry I wasn't clear about it.

Reply:
That would also work, I just wanted to avoid we send both request all the time if anything changes.
I can use the permission check but `this.isSharedWithChanged` would still needed if I don't want to send the `PUT` request if ony the credential name is chaged for example.

Reply:
Added it